tools/c7n_azure/c7n_azure/resources/sqldatabase.py

```diff
@@ -30,3 +39,176 @@ class resource_type(ArmResourceManager.resource_type):
+
+    def list_backup_retention_policies(self, i):
+        client = self.manager.get_client()
+        list_operation = getattr(client, self.operations_property).list_by_database
+
+        resource_group_name = i.get('resourceGroup')
+        server_name = BackupRetentionPolicyFilter.get_sql_server_name(
+            i[ChildResourceQuery.parent_key])
+        database_name = i.get('name')
+
+        list_response = list_operation(
+            resource_group_name, server_name, database_name)
+        return list_response
+
+    def __call__(self, i):
+        policies = self.list_backup_retention_policies(i)
+        return self.filter_with_policies(i, policies)
+
+    def filter_with_policies(self, i, policies):
+        raise NotImplementedError()
+
+    def perform_op(self, a, b):
+        op = BackupRetentionPolicyFilter.number_ops[self.data.get('op', 'eq')]
+        return op(a, b)
+
+    @staticmethod
+    def get_sql_server_name(sql_server_id):
+        parsed = parse_resource_id(sql_server_id)
+        return parsed.get('name')
+
+
+@SqlDatabase.filter_registry.register('short-term-backup-retention-policy')
+class ShortTermBackupRetentionPolicyFilter(BackupRetentionPolicyFilter):
+
+    schema = type_schema(
+        'short-term-backup-retention-policy',
+        required=['retention-period-days'],
+        **{
+            'op': {'enum': list(BackupRetentionPolicyFilter.number_ops.keys())},
+            'retention-period-days': {'type': 'number'},
+        },
+    )
+
+    def __init__(self, data, manager=None):
+        super(ShortTermBackupRetentionPolicyFilter, self).__init__(
+            'backup_short_term_retention_policies', data, manager)
+        self.retention_period_days_limit = self.data.get('retention-period-days')
+
+    def filter_with_policies(self, i, policies):
+        try:
+            for policy in policies:
+                actual_retention_days = policy.get('retention_days')
+                return self.perform_op(actual_retention_days, self.retention_period_days_limit)
+        except CloudError as e:
+            # TODO: is there a way to figure this out before trying to iterate through policies?
+            if e.status_code == 404:
+                return self.filter_without_policies(i)
+            else:
+                raise e
+
+    def filter_without_policies(self, i):
+        # without any backup retention policies, this is effectively 0 days of retention
+        return self.perform_op(0, self.retention_period_days_limit)
+
+
+@SqlDatabase.filter_registry.register('long-term-backup-retention-policy')
+class LongTermBackupRetentionPolicyFilter(BackupRetentionPolicyFilter):
+
+    @enum.unique
+    class BackupType(enum.Enum):
+        weekly = 'weekly'
+        monthly = 'monthly'
+        yearly = 'yearly'
+
+        def __str__(self):
+            return self.value
+
+    @enum.unique
+    class RetentionPeriodUnits(enum.Enum):
+        days = ('days', 'D')
```

Comment:
Wondering if we should move retention period operations to a separate helper class.
It might be useful for Backup Vaults (VM backups) and other backups in Azure.